ocaml/xapi/repository_helpers.ml

```diff
@@ -191,20 +191,20 @@ module Pkg = struct
      *)
     let normalize v =
       let split_letters_and_numbers s =
```

Comment:
This parsing looks very fragile and should probably be reimplemented according to official docs on version strings instead of guessing based on number/letter splits. Also it is likely to reach a different conclusion than RPM/Yum would, so it would be best to ask Yum/RPM about this (there is an rpmvercmp executable if we have to use it) rather than reimplementing this ourselves.

For now lets just fix the regex parts though.

Reply:
I agree that this looks fragile but also with focusing on replacing `Re.Str`.